packages/client/src/components/ModuleTree/Module/Module.tsx

```diff
@@ -1,25 +1,109 @@
 import React from 'react';
-import { Module as EModule } from '@codement/api';
-import classnames from 'classnames';
-import { Icon } from '@codement/ui';
-import Hex from './hex.svg';
-import styles from './Module.module.css';
+import styled from 'styled-components';
+
+import { Module as EModule, UserModule, UserConcept, ModuleType } from '@codement/api/types';
+import { theme } from '@codement/ui';
+import { ModuleIcon } from './Icons/Icon';
 
 export interface ModuleProps {
-  module: EModule
+  module: EModule,
+  myModule?: UserModule,
+  myConcepts?: UserConcept[]
 }
 
+const ModuleContainer = styled.div`
+  text-align: center;
+  margin: 1.5rem;
+  color: ${props => props.theme.colors.primary['500']};
+
+`;
+
+const IconContainer = styled.div`
+  position: relative;
+  display: flex;
+  height: 3rem;
+  width: 3rem;
+  margin-right: auto;
+  margin-left: auto;
+  & svg{
+    position: absolute;
+    left: 50%;
+    top: 50%;
+    transform: translate(-50%, -50%);
+  };
+  & div {
+    display:flex;
+    position: absolute;
+    left: 85%;
+    top: 20%;
+    transform: translate(-50%, -50%);
+    & span{
+      font-size: 0.5rem;
+      color: ${props => props.theme.colors.white};
+      z-index: 100;
+    };
+  }
+
+`;
+
+const ModuleNameWrapper = styled.span<{isLocked: boolean}>`
+  color: ${props => (props.isLocked ? props.theme.colors.grey['700'] : props.theme.colors.grey['900'])};
+  font-size: 0.75rem;
+
+`;
+
 export const Module: React.FC<ModuleProps> = ({
-  module
-}) => <div className={styles.module}>
-  <div className="relative">
-    <Hex
-      className={classnames('h-12 stroke-2', {
-        // TODO: Add completed to module
-        completed: false
-      })}
-    />
-    <Icon icon="fire" size={4} />
-  </div>
-  <span className="text-xs">{module.name}</span>
-</div>;
+  module,
+  myModule,
+  myConcepts
+}) => {
+  const isCompleted = myModule?.completedAt;
+  const isOpen = myModule && !myModule.completedAt;
+  const isLocked = !myModule;
+
+  return <ModuleContainer>
+    <IconContainer>
+      <ModuleIcon
+        icon={module.type === ModuleType.Lesson ? 'hexagon' : 'octagon'}
+        size={3}
+        // eslint-disable-next-line no-nested-ternary
+        color={isCompleted ? theme.colors.secondary['500'] : isOpen ? theme.colors.primary['400'] : theme.colors.white}
+        stroke={isLocked ? theme.colors.grey['500'] : undefined}
+      />
+
+      <ModuleIcon
+        icon="fire"
```

Comment:
What is the fire for here?

Reply:
I should replace "fire" with module.icon. 
This PR has dependency on the PR #220 styled-component Icon, once that PR is merged, I will modify this PR.